#! /bin/sh /usr/share/dpatch/dpatch-run
## 92_ocsp.dpatch by glandium@debian.org
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Patches from bz433594 and bz#433386

@DPATCH@
diff -ruN nss-3.12.0~rc3/mozilla/security/nss/lib/certhigh/ocsp.c /tmp/mozilla/security/nss/lib/certhigh/ocsp.c
--- nss-3.12.0~rc3/mozilla/security/nss/lib/certhigh/ocsp.c	2008-02-13 16:29:12.000000000 +0100
+++ /tmp/mozilla/security/nss/lib/certhigh/ocsp.c	2008-05-28 20:03:11.000000000 +0200
@@ -39,7 +39,7 @@
  * Implementation of OCSP services, for both client and server.
  * (XXX, really, mostly just for client right now, but intended to do both.)
  *
- * $Id: ocsp.c,v 1.50 2008/02/13 15:29:12 kaie%kuix.de Exp $
+ * $Id: ocsp.c,v 1.50.6.1 2008/05/28 18:03:11 kaie%kuix.de Exp $
  */
 
 #include "prerror.h"
@@ -1516,10 +1516,11 @@
 SECStatus
 CERT_DestroyOCSPCertID(CERTOCSPCertID* certID)
 {
-    if (certID->poolp) {
+    if (certID && certID->poolp) {
 	PORT_FreeArena(certID->poolp, PR_FALSE);
 	return SECSuccess;
     }
+    PORT_SetError(SEC_ERROR_INVALID_ARGS);
     return SECFailure;
 }
 
diff -ruN nss-3.12.0~rc3/mozilla/security/nss/lib/libpkix/pkix/checker/pkix_ocspchecker.c /tmp/mozilla/security/nss/lib/libpkix/pkix/checker/pkix_ocspchecker.c
--- nss-3.12.0~rc3/mozilla/security/nss/lib/libpkix/pkix/checker/pkix_ocspchecker.c	2008-03-26 19:49:00.000000000 +0100
+++ /tmp/mozilla/security/nss/lib/libpkix/pkix/checker/pkix_ocspchecker.c	2008-05-28 20:05:33.000000000 +0200
@@ -288,7 +288,11 @@
                 PKIX_OCSPRESPONSEGETSTATUSFORCERTFAILED);
 
 cleanup:
-        if (!passed && cid) {
+        if (!passed && cid && cid->certID && !cid->certIDWasConsumed) {
+                /* We still own the certID object, which means that 
+                 * it did not get consumed to create a cache entry.
+                 * Let's make sure we create one.
+                 */
                 PKIX_Error *err;
                 err = PKIX_PL_OcspCertID_RememberOCSPProcessingFailure(
                         cid, plContext);
